src/app/ui/components/items/item.tsx

```diff
@@ -0,0 +1,137 @@
+import { type ComponentProps, KeyboardEvent } from 'react';
+
+import { sva } from 'leather-styles/css';
+import { BoxProps, styled } from 'leather-styles/jsx';
+
+import { createStyleContext } from '@app/ui/utils/style-context';
+
+const item = sva({
+  slots: ['root', 'content', 'text', 'caption'],
+  base: {
+    root: {
+      bg: 'accent.background-primary',
+      height: 'auto',
+      userSelect: 'none',
+      p: 'space.03',
+      width: '100%',
+    },
+    content: {
+      width: '100%',
+    },
+    text: {
+      color: 'accent.text-primary',
+      fontWeight: 500,
+      textStyle: 'label.02',
+    },
+    caption: {
+      color: 'accent.text-subdued',
+    },
+  },
+  variants: {
+    isDisabled: {
+      true: {
+        root: {
+          _active: { bg: 'unset' },
+          _focus: { border: 'unset' },
+          _hover: { bg: 'unset' },
+          cursor: 'not-allowed',
+        },
+        content: {},
+        text: {
+          color: 'accent.non-interactive',
+        },
+        caption: {
+          color: 'accent.non-interactive',
+        },
+      },
+    },
+    isPressable: {
+      true: {
+        root: {
+          _active: {
+            bg: 'accent.component-background-pressed',
+          },
+          _focus: {
+            border: '2px solid {focus}',
+          },
+          _hover: {
+            bg: 'accent.component-background-hover',
+          },
+        },
+      },
+    },
+    isSelected: {
+      true: {
+        root: {
+          _active: {
+            bg: 'unset',
+          },
+          _focus: {
+            border: 'unset',
+          },
+          _hover: {
+            bg: 'unset',
+          },
+        },
+      },
+    },
+  },
+  compoundVariants: [
+    {
+      isDisabled: true,
+      isPressable: true,
+      css: {
+        root: {
+          _active: { bg: 'unset' },
+          _focus: { border: 'unset' },
+          _hover: { bg: 'unset' },
+          cursor: 'not-allowed',
+        },
+      },
+    },
+  ],
+});
+
+const { withProvider, withContext } = createStyleContext(item);
+
+const RootBase = withProvider('div', 'root');
+
+interface RootProps extends ComponentProps<'div'> {
+  isDisabled?: boolean;
+  isPressable?: boolean;
+  isSelected?: boolean;
+  onClick?(): void;
+}
+function Root({ isDisabled, isPressable, isSelected, onClick, ...props }: RootProps & BoxProps) {
+  const isRoleButton = !!onClick;
+
+  function onKeyDown(e: KeyboardEvent<HTMLInputElement>) {
+    if (!isRoleButton) return;
+    if (e.key === 'enter' || e.key === ' ') {
+      e.preventDefault();
+      onClick();
+    }
+  }
```

Comment:
Why is this needed? If the element's a button it should behave this way by default.

Reply:
This is not always a button. I am adding this here after referencing how to add accessibility to a div so that it can be either a div or a button.